### PR TITLE
SCI Embedding of git metadata

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
* Add `DD_TRACE_GIT_METADATA_ENABLED` config env var, with a default of `true`
* Attempt to read `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA` env vars. 
* Attempt to read `git.repository_url` and `git.commit.sha` from `DD_TAGS`.
* If they're present, set these values as tags to the _first_ span in a trace. 

### Motivation
Allow some products to provide source code related features. 

### Additional Notes
Unrelated fix: warp `esbuild.spec.js` test in a `describe`.

### CODE REVIEW

File: integration-tests/esbuild.spec.js [hunk 1]
```diff
@@ -8,24 +8,28 @@ const path = require('path')
...
+    }
   })
-} catch (err) {
-  // eslint-disable-next-line no-console
-  console.error(err)
-  process.exit(1)
-} finally {
-  process.chdir(CWD)
-}
+})
```
Comment (juan-fernandez) [inline integration-tests/esbuild.spec.js:35]: unrelated fix. **Unconfirmed** that this is necessary

File: integration-tests/sci-embedding.spec.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  curlAndAssertMessage
+} = require('./helpers')
+const path = require('path')
+const { assert } = require('chai')
+const { SCI_COMMIT_SHA, SCI_REPOSITORY_URL } = require('../packages/dd-trace/src/constants')
+
+const DUMMY_GIT_SHA = '13851f2b092e97acebab1b73f6c0e7818e795b50'
+const DUMMY_REPOSITORY_URL = 'git@github.com:DataDog/sci_git_example.git'
+
+describe('sci embedding', function () {
```
Comment (rochdev) [inline integration-tests/sci-embedding.spec.js:16]: Does this need to be an integration test?

File: packages/dd-trace/src/config.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const coalesce = require('koalas')
 const tagger = require('./tagger')
 const { isTrue, isFalse } = require('./util')
 const uuid = require('crypto-randomuuid')
+const { GIT_REPOSITORY_URL, GIT_COMMIT_SHA } = require('./plugins/util/tags')
```
Comment (rochdev) [inline packages/dd-trace/src/config.js:12]: It looks like these are expected to be set by users? If that's the case they should probably be moved to `ext/tags`.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: integration-tests/esbuild.spec.js (hunk 1 above)
Comment (juan-fernandez) [reply integration-tests/esbuild.spec.js:35]: it's needed

File: packages/dd-trace/src/config.js (hunk 3 above)
Comment (juan-fernandez) [reply packages/dd-trace/src/config.js:12]: yeah that makes sense. I'll do this in another PR though, because there are multiple files to change and I don't want to make this harder to review

File: integration-tests/sci-embedding.spec.js (hunk 2 above)
Comment (juan-fernandez) [reply integration-tests/sci-embedding.spec.js:16]: I thought of integration because of the usage of "external" env vars, but this is done in other unit tests. I'll migrate 👍

File: packages/dd-trace/src/git_metadata_tagger.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const { SCI_COMMIT_SHA, SCI_REPOSITORY_URL } = require('./constants')
+
+class GitMetadataTagger {
+  constructor (config) {
+    this._config = config
+  }
+
+  tagGitMetadata (spanContext) {
+    if (this._config.gitMetadataEnabled) {
+      spanContext._trace.tags[SCI_COMMIT_SHA] = this._config.commitSHA
```
Comment (juan-fernandez) [inline packages/dd-trace/src/git_metadata_tagger.js:10]: this trace tags are only added to the local root in `format.js` via `extractChunkTags`

**Review by rochdev — APPROVED** [review]

**Review by sashacmc — APPROVED** [review]
LGTM!
